### DISCUSSION TITLE
Group Wazuh alerts into groups

### DISCUSSION THREAD

**Discussion by alvarocl1981** [wazuh/wazuh#25280] [General]
Custom labels filter button in Threat Intelligence - Vulnerability detection do not work. We have tried to use TAGs label in agent.conf for the same purpose without success neither. Is there a way to filter by group of agents in Vulnerability detection section?

**Comment by jftuduri** [comment]
Hi @alvarocl1981!

At the moment there is not a direct way to filter vulnerability data by agent group. But there are a couple of workarounds that might be useful depending on the use case:

[Labels](https://documentation.wazuh.com/current/user-manual/agent/agent-management/labels.html) can be used in the Events tab. 
They can be filtered with something like this: `agent.labels.test_label:custom-group`
In this case the label name is `test_label` and the filtered value is `custom-group`.
I had added this configuration to the agent:
```
<labels>
  <label key="test_label">custom-group</label>
</labels>
```
![image](https://github.com/user-attachments/assets/f61d5f93-c9a0-451f-9c78-d0a896a4981e)

As mentioned, that option only works for the Events tab.
For the Dashboard and Inventory tabs an option is to use a custom filter with the desired agent IDs.
For example, the following query shows only the data for agents `001` and `003`:
```
{
  "query": {
    "terms": {
      "agent.id": [
        "001",
        "003"
      ]
    }
  }
}
```
It uses the terms query: https://opensearch.org/docs/latest/query-dsl/term/terms/

To use this you need to `Add filter` and then `Edit as query DSL`: 
![image](https://github.com/user-attachments/assets/147ee38a-c23b-447d-94dd-86adfa6fecfa)

Then paste the filter and, optionally give it a name:
![image](https://github.com/user-attachments/assets/d6034f1a-fc78-4cd6-b2a2-625a3a8bea25)

Regards!

**Comment by Damian-Mangold** [comment]
Closed, more than 7 days of inactivity.
